source/credentials_provider_sts_web_identity.c

```diff
@@ -982,13 +984,15 @@ static struct sts_web_identity_parameters *s_parameters_new(struct aws_allocator
      * check config profile if either region, role_arn or token_file_path or role_session_name is not resolved from
      * environment variable. Role session name can also be generated by us using uuid if not found from both sources.
      */
-    struct aws_profile_collection *config_profile = NULL;
+    struct aws_profile_collection *config_profile = aws_profile_collection_acquire(config_profile_collection_cached);
     struct aws_string *profile_name = NULL;
     const struct aws_profile *profile = NULL;
     bool get_all_parameters =
         (region && region->len && role_arn && role_arn->len && token_file_path && token_file_path->len);
     if (!get_all_parameters) {
-        config_profile = s_load_profile(allocator);
+        if (!config_profile) {
+            config_profile = s_load_profile(allocator);
+        }
```

Comment:
trivial/debatable: similar to earlier feedback, I feel like  newcomer to this code would have an easier time reading something like 
```
if (config_profile_collection_cached) {
    /* Use cached profile collection */
    config_profile = aws_profile_collection_acquire(config_profile_collection_cached);
} else {
    /* Load profile collection from files */
    config_profile = s_load_profile(allocator);
}
```

Reply:
do we need to check the results of `s_load_profile()`?

oh ... a bunch of lines later we catch the error, but it prints out the same message as if profile-load succeeded and but get-profile-by-name failed

maybe better to `goto on_finish;` earlier?

Reply:
Thanks, updated.